### DISCUSSION TITLE
How to use dataloaders with graphql-ws subscriptions?

### DISCUSSION THREAD

**Discussion by 97amarnathk** [enisdenjo/graphql-ws#198] [Q&A] [answered]
I am trying to use dataloaders for caching execution result of my subscription.
However I am not able to find a way to invalidate the dataloader cache after each execution of graphql.

I was planning to write a custom `execute`  hook, where I will invalidate dataloader in the end. 

I noticed that graphql-ws does not call `execute`  for each subscription, and instead it uses `subscribe` provided by graphql-js, which uses the default `execute`.

Is there an alternative to defining my on subscribe method  ? That would be a little complicated, and not useful since I just need to replace the `execute` method being used.

I see that https://github.com/graphql/graphql-js/pull/3071 tries to do the same. Any other alternatives till this is merged ?

**Comment by n1ru4l** [comment, marked as answer]
You can provide a custom `subscribe` function as an option. https://github.com/enisdenjo/graphql-ws/blob/784ea15c86b42657af77a6e03956fc15fd26dd0c/src/server.ts#L162-L170

Right now `execute` function is an implementation detail of `subscribe` and thus there is no option to replace it. However, writing your own is pretty straight-forward: https://github.com/dotansimha/envelop/pull/183/files#diff-8172edfaa44ae919cd25a7283b36668248f1484041827d70e66da3aacccde461R7-R49
